indi-qhy/qhy_ccd.cpp

```diff
@@ -1778,6 +1840,50 @@ bool QHYCCD::ISNewSwitch(const char *dev, const char *name, ISState *states, cha
             IDSetSwitch(&AMPGlowSP, nullptr);
             return true;
         }
+
+        //////////////////////////////////////////////////////////////////////
+        /// Overscan Area
+        //////////////////////////////////////////////////////////////////////
+        //NEW CODE - Add support for overscan/calibration area 
+        else if (!strcmp(OverscanAreaSP.name, name))
+        {   
+            
+            IUUpdateSwitch(&OverscanAreaSP, states, names, n);
+            bool isIgnored = (OverscanAreaS[INDI_ENABLED].s == ISS_OFF); //Overscan Area included switch is 'off', thus excluded
+            
+            if (isIgnored == IgnoreOverscanArea)
+                return true;
+
+            IgnoreOverscanArea = isIgnored;
+            
+            OverscanAreaS[INDI_DISABLED].s = IgnoreOverscanArea ? ISS_ON : ISS_OFF; 
+            OverscanAreaS[INDI_ENABLED].s = IgnoreOverscanArea ? ISS_OFF : ISS_ON;
+
+            OverscanAreaSP.s = IgnoreOverscanArea ? IPS_IDLE : IPS_OK;
+
+            IDSetSwitch(&OverscanAreaSP, nullptr);
+
+            if (IgnoreOverscanArea)
+            {
+                //Image Info 
+                SetCCDParams(effectiveROI.subW, effectiveROI.subH, PrimaryCCD.getBPP(), PrimaryCCD.getPixelSizeX(), PrimaryCCD.getPixelSizeX());
+
+                //Image Settings
+                //The true frame origin is at (effectiveROI.subX ,effectiveROI.subY), need to correct for this offset when taking exposure or streaming. 
+                UpdateCCDFrame(0 ,0, effectiveROI.subW, effectiveROI.subH);
+            
+            } else { 
+                //Image Info 
+                SetCCDParams(sensorROI.subW, sensorROI.subH, PrimaryCCD.getBPP(), PrimaryCCD.getPixelSizeX(), PrimaryCCD.getPixelSizeX());
+                //Image Settings
+                UpdateCCDFrame(sensorROI.subX ,sensorROI.subY, sensorROI.subW, sensorROI.subH);
+            }
+
+            LOGF_INFO("The overscan area is %s now. The effective frame starts at coordinates (%u, %u) ", IgnoreOverscanArea ? "ignored" : "included", IgnoreOverscanArea ? effectiveROI.subX : 0, IgnoreOverscanArea ? effectiveROI.subY : 0);
+            
+            IDSetSwitch(&OverscanAreaSP, nullptr);
```

Comment:
Why are you setting the OverscanAreaSP switch twice here? One should be enough the at the end.

Reply:
Yes, good find. That was a copy-and-paste leftover,  thanks for seeing it.  It is also updated in the last commit.